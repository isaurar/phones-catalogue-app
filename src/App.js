import React from 'react';
import { Route } from 'react-router-dom';
import { makeStyles } from '@material-ui/core/styles';
import Container from '@material-ui/core/Container';
import AppBar from '@material-ui/core/AppBar';
import Typography from '@material-ui/core/Typography';
import PhonesListContainer from './modules/phones-list/PhonesListContainer';
import PhoneDetailsContainer from './modules/phone-details/PhoneDetailsContainer';

const useStyles = makeStyles({
  appBar: {
    padding: '1rem'
  },
  topBarSeparator: {
    marginBottom: '5rem'
  }
});

function App() {
  const classes = useStyles();
  return (
    <Container fixed>
      <AppBar className={classes.appBar}>
        <Typography variant="h6">Phones Catalogue</Typography>
      </AppBar>
      <div className={classes.topBarSeparator} />
      <Route path="/phones-list" component={PhonesListContainer}></Route>
      <Route path="/phone-details/:id" component={PhoneDetailsContainer}></Route>
    </Container>
  );
}

export default App;
