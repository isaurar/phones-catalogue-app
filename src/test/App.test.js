import React from 'react';
import App from '../App.js';
import TestRenderer from 'react-test-renderer';
import {BrowserRouter as Router} from 'react-router-dom';

test('Application should render the phone catalogue', () => {
    const rendered = TestRenderer.create(
        <Router>
            <App />
        </Router>
    ).toJSON();
    expect(rendered).toMatchSnapshot();
});