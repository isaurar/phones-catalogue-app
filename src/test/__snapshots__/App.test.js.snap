// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Application should render the phone catalogue 1`] = `
<div
  className="MuiContainer-root MuiContainer-fixed MuiContainer-maxWidthLg"
>
  <header
    className="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary makeStyles-appBar-1 mui-fixed MuiPaper-elevation4"
  >
    <h6
      className="MuiTypography-root MuiTypography-h6"
    >
      Phones Catalogue
    </h6>
  </header>
  <div
    className="makeStyles-topBarSeparator-2"
  />
</div>
`;
